### ISSUE TITLE
Typescript support

### ISSUE THREAD

**Issue by ChaudhryAtif** [nickpeihl/leaflet-sidebar-v2#12]
Can Typerscript support be added for the sidebar? The parent repo has a file that can be dropped-in, or used as a starting point.
[leaflet-sidebar.d.ts](https://github.com/Turbo87/sidebar-v2/blob/master/typescript/leaflet-sidebar.d.ts)

Currently gives me the following error:

> Property 'sidebar' does not exist on type 'typeof control'.

When trying to do the following (using autopan example):
```
L.control.sidebar('sidebar', { autopan: true })
    .addTo(map)
    .open('autopan');
```

And if possible, please add an angular 2+ example.

**Comment by nickpeihl** [comment]
Hi @ChaudhryAtif 

I am not familiar with TypeScript or Angular. Would you like to [submit pull requests](https://help.github.com/articles/about-pull-requests/) for including TypeScript definitions and an Angular example?

**Comment by ChaudhryAtif** [comment]
The parent repo has a typing file, but this fork is ahead of the parent repo, So typing file will need to be modified, but I do not have any experience in writing typings for a JS project. If someone can write the typing file, I can probably add an Angular example.

**Comment by noerw** [comment]
Yeah, a little bit. I can look into it.
Can't help out with angular 2 though

**Comment by ChaudhryAtif** [comment]
I started looking into it as well. I will keep you guys posted on progress. Using [leaflet-sidebar.d.ts](https://github.com/Turbo87/sidebar-v2/blob/master/typescript/leaflet-sidebar.d.ts) as base.

**Comment by noerw** [comment]
fixed in 9873c9036195625ad8cc5d1d8f52045755b8889d

**Comment by ChaudhryAtif** [comment]
@nickpeihl @noerw Will the version get a bump? B/c npm install 3.0.0 doesn't seem to give the latest changes.

**Comment by noerw** [comment]
I just released v3.0.1

**Comment by cwgrc2** [comment]
Can anyone here please give an example on actually how to use this in Typescript Angular?  A little late to the game here sorry.  I have been trying to integrate leaflet-sidebar-v2 for two days now with no luck.  I tried '@runette/ngx-leaflet-sidebar' Angular wrapper but that requires a certain version of Angular to work.

Thanks!

**Comment by ChaudhryAtif** [comment]
What have you tried? Since `leaflet-sidebar.d.ts` is inside the project, you shouldn't have to do a lot other than importing it. 

Hopefully this helps:
```
// package.json
npm install -S leaflet leaflet-sidebar-v2
npm install -D @types/leaflet

// html
<div id="sidebar" class="leaflet-sidebar collapsed">
    ...
</div>

// controller
import * as L from 'leaflet';
import 'leaflet-sidebar-v2';

getSidebar() {
    return L.control.sidebar({container: 'sidebar', autopan: true});
}
```

**Comment by cwgrc2** [comment]
Thanks for Responding Atif!

I am first taking Paul's Runette's advice of rebuilding all of my
node_modules with the latest of all of our add-on plugins.

Is it your contention above that we do not need the
'runnete/ngx-leaflet-sidebar' add-on and just use the leaflet-sidebar-v2
directly?

If so, I  think I tried that, but I think I tried it without
the @types/leaflet install.  So I will try that again first after the
re-install.

Thanks for your input!

Chris
scubagrant60@gmail.com
